### ISSUE TITLE
Potential bug in update_bias: Unit mismatch between frequency and assignment count

### ISSUE THREAD

**Issue by LianyouII** [gfyddha/FlyLoRA#1]
Hi,I believe I found a logic error in FlyLoRALinear.update_bias in flylora.py.
In the following code block:
    @torch.no_grad()
    def update_bias(self):
        """Update expert bias for load balancing: d_i ← d_i + u·sign(̄c_i - c_i)"""
        delta = torch.sign(self.expected_frequency - self.assignment_count)
        self.d.add_(self.bias_lr * delta)
        self.assignment_count.zero_()
self.expected_frequency is initialized as a probability k/r (e.g., 0.25), whereas self.assignment_count accumulates the raw integer counts of expert selections.Unless the batch size is 1, self.assignment_count for any active expert will typically be an integer, which is greater than self.expected_frequency. As a result, delta becomes -1 for almost all experts (any expert selected at least once), and +1 only for experts with exactly 0 selections.This causes the load balancing to penalize heavily used experts and lightly used experts equally, failing to balance the load effectively when the batch size is large. It seems self.assignment_count should be normalized by the batch size (or self.expected_frequency scaled by the batch size) before the subtraction.
Could you please check this? Thanks

**Comment by gfyddha** [comment]
Hi, thank you for your suggestions. The original example in `flylora.py` had a few minor flaws, and I've updated the file to better align with our implementation in the `peft` library.
